pkg/fanal/analyzer/language/julia/pkg/pkg_test.go

```diff
@@ -0,0 +1,247 @@
+package pkgjl
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	"github.com/aquasecurity/trivy/pkg/fanal/analyzer"
+	"github.com/aquasecurity/trivy/pkg/fanal/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_juliaAnalyzer_Analyze(t *testing.T) {
+	tests := []struct {
+		name string
+		dir  string
+		want *analyzer.AnalysisResult
+	}{
+		{
+			name: "happy path",
+			dir:  "testdata/happy",
+			want: &analyzer.AnalysisResult{
+				Applications: []types.Application{
+					{
+						Type:     types.Julia,
+						FilePath: "Manifest.toml",
+						Libraries: []types.Package{
+							{
+								ID:        "ade2ca70-3891-5945-98fb-dc099432e06a",
```

Comment:
I'm sorry I'm probably missing some context. Do we need to use UUID here? Can the combination `name + version` be non-unique?

Reply:
There are cases when a file contains two dependencies with same name and version, but with different UUIDs - https://github.com/aquasecurity/trivy/pull/5635#issuecomment-1843676898
We thought it would be better to use UUIDs to separate dependencies, like Julia does.

Reply:
I'd like to see why it is allowed. Is there any document?

Reply:
@Octogonapus Do you have any documents regarding this case?

Reply:
See https://pkgdocs.julialang.org/v1/toml-files/#Multiple-packages-with-the-same-name